client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx

```diff
@@ -133,6 +135,7 @@ const ScreenMain = ( { onSelect, onContinueClick, recordTracksEvent }: Props ) =
 						</>
 					</NavigatorItemGroup>
 				</HStack>
+				<Survey userHasEngaged={ userHasEngaged } />
```

Comment:
[nit] We can simply check the value of `userHasEngaged` here and determine whether to render the survey but the current way is also fine to me 🙂